files/classes/submission.py

```diff
@@ -47,26 +46,23 @@ class Submission(CreatedBase):
 	body = Column(Text)
 	body_html = Column(Text)
 	flair = Column(String)
-	ban_reason = Column(String)
 	embed_url = Column(String)
-	filter_state = Column(String, nullable=False)
 	task_id = Column(Integer, ForeignKey("tasks_repeatable_scheduled_submissions.id"))
 
 	# Visibility states here
 	state_user_deleted_utc = Column(DateTime(timezone=True), nullable=True) # null if it hasn't been deleted by the user
-	# TBD: state_mod
-	# TBD: state_mod_set_by
-	# TBD: state_report
+	state_mod = Column(Enum(StateMod), default=StateMod.Filtered, nullable=False) # default to Filtered just to partially neuter possible exploits
+	state_mod_set_by = Column(String, nullable=True) # This should *really* be a User.id, but I don't want to mess with the required refactoring at the moment - it's extra hard because it could potentially be a lot of extra either data or queries
+	state_report = Column(Enum(StateReport), default=StateReport.Unreported, nullable=False)
 
-	Index('fki_submissions_approver_fkey', is_approved)
 	Index('post_app_id_idx', app_id)
-	Index('subimssion_binary_group_idx', is_banned, state_user_deleted_utc, over_18)
-	Index('submission_isbanned_idx', is_banned)
+	Index('subimssion_binary_group_idx', state_mod, state_user_deleted_utc, over_18)
+	Index('submission_state_mod_idx', state_mod)
```

Comment:
I think that this index could be deleted btw.

Reply:
Probably; I don't really want to mess with it right now, though, this is already doing too much >_<

I've added a task at #607 to take care of this stuff though.